src/dbc.rs

```diff
@@ -330,89 +525,117 @@ mod tests {
                 );
             }
         }
 
         // Valid mint signatures BUT signing wrong message
         for _ in 0..n_wrong_msg_sigs.coerce() {
             if let Some(input) = repeating_inputs.next() {
-                let wrong_msg_sig = genesis_node.key_manager.sign(&Hash([0u8; 32])).unwrap();
-                let wrong_msg_mint_sig = genesis_node
-                    .key_manager
-                    .public_key_set()
-                    .unwrap()
+                let wrong_msg_sig = mint_node.key_manager.sign(&Hash([0u8; 32])).unwrap();
+                let wrong_msg_mint_sig = mint_node
+                    .key_manager()
+                    .public_key_set()?
                     .combine_signatures(vec![wrong_msg_sig.threshold_crypto()])
                     .unwrap();
 
-                fuzzed_transaction_sigs
-                    .insert(input.spend_key(), (genesis_key, wrong_msg_mint_sig));
+                fuzzed_transaction_sigs.insert(
+                    input.key_image.into(),
+                    (
+                        mint_node.key_manager.public_key_set()?.public_key(),
+                        wrong_msg_mint_sig,
+                    ),
+                );
             }
         }
 
         // Valid mint signatures for inputs not present in the transaction
         for _ in 0..n_extra_input_sigs.coerce() {
-            fuzzed_transaction_sigs
-                .insert(rand::random::<SpendKey>(), (genesis_key, mint_sig.clone()));
+            fuzzed_transaction_sigs.insert(
+                Default::default(),
+                (
+                    mint_node.key_manager().public_key_set()?.public_key(),
+                    mint_sig.clone(),
+                ),
+            );
         }
 
         let dbc = Dbc {
             content: fuzzed_content,
-            transaction: reissue_share.dbc_transaction,
+            transaction: reissue_share.transaction.clone(),
             transaction_sigs: fuzzed_transaction_sigs,
+            spent_proofs: reissue_share.spent_proofs.clone(), // todo: fuzz spent proofs.
         };
 
-        let id = crate::bls_dkg_id();
-        let key_manager = SimpleKeyManager::new(SimpleSigner::from(id), genesis_key);
-        let validation_res = dbc.confirm_valid(&key_manager);
+        let key_manager = mint_node.key_manager();
+        let validation_res = dbc.confirm_valid(&owner_once.owner_base().secret_key()?, key_manager);
 
-        let dbc_amount = DbcHelper::decrypt_amount(&input_owner, &dbc.content)?;
+        let dbc_owner = dbc
+            .owner_once(&owner_once.owner_base().secret_key()?)?
+            .public_key_blst();
+
+        // Check if commitment in AmountSecrets matches the commitment in tx OutputProof
+        let commitments_match = dbc
+            .confirm_provided_amount_matches_commitment(
+                &owner_once.owner_base().secret_key()?,
+                &fuzzed_amt_secrets,
+            )
+            .is_ok();
```

Comment:
If we move this confirmation into `confirm_valid()`, we should remember to remove this

Reply:
commented in #127